audbackend/core/backend.py

```diff
@@ -439,14 +444,41 @@ def latest_version(
         vs = self.versions(path)
         return vs[-1]
 
+    def _legacy_split_ext(
+            self,
+            name: str,
+    ) -> typing.Tuple[str, str]:
+        r"""Split name into basename and extension."""
+        ext = None
+        for custom_ext in self._legacy_extensions:
+            # check for custom extension
+            # ensure basename is not empty
+            if self._legacy_file_structure_regex:
+                pattern = rf'\.({custom_ext})$'
+                match = re.search(pattern, name[1:])
+                if match:
+                    ext = match.group(1)
+            elif name[1:].endswith(f'.{custom_ext}'):
+                ext = custom_ext
+        if ext is None:
+            # if no custom extension is found
+            # use last string after dot
+            ext = audeer.file_extension(name)
+
+        base = audeer.replace_file_extension(name, '', ext=ext)
+
+        if ext:
+            ext = f'.{ext}'
+
+        return base, ext
+
     def _ls(
             self,
             path: str,
-    ) -> typing.List[typing.Tuple[str, str]]:  # pragma: no cover
-        r"""List all files under (sub-)path.
+    ) -> typing.List[str]:  # pragma: no cover
+        r"""List all files under sub-path.
 
         * If path does not exist an error should be raised
-        * If path ends on `/` it is a sub-path
```

Comment:
Is this no longer true?

Reply:
I removed it because we always pass a sub-path now and in the line above the docstring now says "List all files under sub-path."

Reply:
However, we should mention that an empty list should be returned for an empty repository. I now changed the comment to:

```python
    def _ls(
            self,
            path: str,
    ) -> typing.List[str]:  # pragma: no cover
        r"""List all files under sub-path.

        If ``path`` is ``'/'`` and no files exist on the repository,
        an empty list should be returned
        Otherwise,
        if ``path`` does not exist or no files are found under ``path``,
        an error should be raised.

        """
        raise NotImplementedError()
``